air-quality-ui/system_tests/ui/summary_page/summary_page_table_visuals.spec.ts

```diff
@@ -0,0 +1,228 @@
+            measurement_base_time: '2024-07-08T03:00:00Z',
+            no2: {
+              mean: { aqi_level: CaseAQI2.aqiLevel, value: CaseAQI2.no2 },
+            },
+            o3: {
+              mean: { aqi_level: CaseAQI3.aqiLevel, value: CaseAQI3.o3 },
+            },
+            pm2_5: {
+              mean: { aqi_level: CaseAQI4.aqiLevel, value: CaseAQI4.pm2_5 },
+            },
+            pm10: {
+              mean: { aqi_level: CaseAQI5.aqiLevel, value: CaseAQI5.pm10 },
+            },
+            so2: {
+              mean: { aqi_level: CaseAQI6.aqiLevel, value: CaseAQI6.so2 },
+            },
+          }),
+        ],
+      )
+    })
+    test('Default Toggle: Highlight all AQI values', async ({
+      summaryPage,
+    }) => {
+      const expectedTableColours: string[][] = [
+        [
+          // AQI Level
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Diff
+          // pm2.5
+          Colours.aqi3, // Forecast
+          Colours.aqi4, // Measured
+          Colours.notColoured, // Time
+          // pm10
+          Colours.aqi4, // Forecast
+          Colours.aqi5, // Measured
+          Colours.notColoured, // Time
+          // no2
+          Colours.aqi1, // Forecast
+          Colours.aqi2, // Measured
+          Colours.notColoured, // Time
+          // o3
+          Colours.aqi2, // Forecast
+          Colours.aqi3, // Measured
+          Colours.notColoured, // Time
+          // so2
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Time
+        ],
+      ]
+
+      await summaryPage.assertGridAttributes('colours', expectedTableColours)
+    })
+    test('Toggle OFF: Highlight primary AQI values', async ({
+      summaryPage,
+    }) => {
+      const expectedTableColours: string[][] = [
+        [
+          // AQI Level
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Diff
+          // pm2.5
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // pm10
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // no2
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // o3
+          Colours.notColoured, // Forecast
+          Colours.notColoured, // Measured
+          Colours.notColoured, // Time
+          // so2
+          Colours.aqi1, // Forecast
+          Colours.aqi6, // Measured
+          Colours.notColoured, // Time
+        ],
+      ]
+      await summaryPage.highlightValuesToggle.click()
+      await summaryPage.assertGridAttributes('colours', expectedTableColours)
+    })
+  })
+  test.describe('No Data', () => {
```

Comment:
```suggestion
  })
  
  test.describe('No Data', () => {
```
it helps readability if there are some empty lines at between the end of one test and what comes next, could you add some?